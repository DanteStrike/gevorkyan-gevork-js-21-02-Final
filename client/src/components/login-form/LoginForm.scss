@import "src/assets/variables";

.login-form {
  width: 380px;

  &__last {
    margin-bottom: 0;
  }

  &__link {
    color: var(--secondory-font-color);
    text-align: center;
  }

  .ant-form-item-label {
    padding: 0;
    font-weight: 700;
  }

  .ant-form-item-label > label {
    color: var(--primary-font-color);
  }
}

@media (max-width: $tabletWidth) {
  .login-form {
    width: 85%;
  }
}
