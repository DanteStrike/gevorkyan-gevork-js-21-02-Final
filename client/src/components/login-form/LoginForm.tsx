import React from 'react';
import './LoginForm.scss';
import {Form, Input} from 'antd';
import Button from '../submit-button/SubmitButton';
import CustomLink from '../custom-link/CustomLink';
import {RoutePath} from '../../enums';
import {ValidateUtils} from '../../utils';
import useAppTranslation from '../../hooks/use-app-translation';

interface ILoginFormProps {
  loading?: boolean;
  onSubmit?: (id: string) => void;
}

const {Item} = Form;

function LoginForm({onSubmit = () => {}, loading}: ILoginFormProps) {
  const {t} = useAppTranslation(`loginForm`);
  const [form] = Form.useForm();

  const handleFormFinish = (filedValues: {id: string}) => {
    onSubmit(filedValues.id);
  };

  return (
    <Form
      className="login-form"
      form={form}
      name="login"
      size="middle"
      layout="vertical"
      onFinish={handleFormFinish}
      initialValues={{
        id: ``,
      }}
    >
      <Item name="id" label="ID:" rules={[ValidateUtils.requireValidatorID]}>
        <Input placeholder={t(`id.placeholder`)} />
      </Item>

      <Item className="login-form__last">
        <Button loading={loading}>{t(`enter`)}</Button>
        <CustomLink className="login-form__link" to={RoutePath.REGISTRATION}>
          {t(`message`)}
        </CustomLink>
      </Item>
    </Form>
  );
}

LoginForm.defaultProps = {
  loading: false,
  onSubmit: () => {},
};

export default React.memo(LoginForm);
