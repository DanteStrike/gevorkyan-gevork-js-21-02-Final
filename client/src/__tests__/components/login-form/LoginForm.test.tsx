import {mount, shallow} from 'enzyme';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-dom';
import React from 'react';
import sleep from '../../mocks/utils';
import LoginForm from '../../../components/login-form/LoginForm';

describe(`Component LoginForm should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot loading test`, () => {
    const wrapper = shallow(<LoginForm loading />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe(`Component LoginForm should work correctly`, () => {
  it(`call submit test`, () => {
    const history = createMemoryHistory();
    const submitHandler = jest.fn(() => {});
    const wrapper = mount(
      <Router history={history}>
        <LoginForm onSubmit={submitHandler} />
      </Router>
    );

    const mockData = {
      id: `anyID`,
    };

    wrapper.children().children().prop(`form`).setFieldsValue(mockData);
    wrapper.update();
    wrapper.find('form').simulate('submit');

    return sleep().then(() => {
      expect(submitHandler).toHaveBeenCalledTimes(1);
      expect(submitHandler).toHaveBeenLastCalledWith(`anyID`);
    });
  });
});
