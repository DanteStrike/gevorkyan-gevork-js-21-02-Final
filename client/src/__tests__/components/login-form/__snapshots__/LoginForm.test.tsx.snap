// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoginForm should render correctly snapshot default test 1`] = `
<ForwardRef(InternalForm)
  className="login-form"
  form={
    Object {
      "__INTERNAL__": Object {
        "itemRef": [Function],
      },
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "getInternalHooks": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "resetFields": [Function],
      "scrollToField": [Function],
      "setFields": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
    }
  }
  initialValues={
    Object {
      "id": "",
    }
  }
  layout="vertical"
  name="login"
  onFinish={[Function]}
  size="middle"
>
  <FormItem
    label="ID:"
    name="id"
    rules={
      Array [
        [Function],
      ]
    }
  >
    <Input
      placeholder="Input ID"
      type="text"
    />
  </FormItem>
  <FormItem
    className="login-form__last"
  >
    <Memo(SubmitButton)
      loading={false}
    >
      Enter
    </Memo(SubmitButton)>
    <Memo(CustomLink)
      className="login-form__link"
      to="/registration"
    >
      Haven't account? Register
    </Memo(CustomLink)>
  </FormItem>
</ForwardRef(InternalForm)>
`;

exports[`Component LoginForm should render correctly snapshot loading test 1`] = `
<ForwardRef(InternalForm)
  className="login-form"
  form={
    Object {
      "__INTERNAL__": Object {
        "itemRef": [Function],
      },
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "getInternalHooks": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "resetFields": [Function],
      "scrollToField": [Function],
      "setFields": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
    }
  }
  initialValues={
    Object {
      "id": "",
    }
  }
  layout="vertical"
  name="login"
  onFinish={[Function]}
  size="middle"
>
  <FormItem
    label="ID:"
    name="id"
    rules={
      Array [
        [Function],
      ]
    }
  >
    <Input
      placeholder="Input ID"
      type="text"
    />
  </FormItem>
  <FormItem
    className="login-form__last"
  >
    <Memo(SubmitButton)
      loading={true}
    >
      Enter
    </Memo(SubmitButton)>
    <Memo(CustomLink)
      className="login-form__link"
      to="/registration"
    >
      Haven't account? Register
    </Memo(CustomLink)>
  </FormItem>
</ForwardRef(InternalForm)>
`;
